deployment/data-streams/utils/txutil/get-tx.go

```diff
@@ -0,0 +1,70 @@
+package txutil
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	goEthTypes "github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+// HasContractAddress is the generic interface for anything
+// that knows which contract address to use.
+type HasContractAddress interface {
+	GetContractAddress() common.Address
+}
+
+// ContractLoader is a function that, given an Environment, chain selector,
+// and contract address, returns the strongly-typed contract “state”
+type ContractLoader[S any] func(
+	e deployment.Environment,
+	chainSelector uint64,
+	contractAddr string,
+) (*S, error)
+
+// ContractMethod is a function that, given a loaded contract state (pointer to S)
+// and some config T, returns a transaction.
+type ContractMethod[S any, T any] func(
+	*S,
+	T,
+) (*goEthTypes.Transaction, error)
+
+// GetTxs is the generic method that:
+//
+// Iterates over chainSelector => []T
+// Loads the contract using the given loader
+// Calls the method function to build the transaction
+// Accumulates them into a slice of PreparedTx.
+func GetTxs[S any, T HasContractAddress](
+	e deployment.Environment,
+	contractType string,
+	configsByChain map[uint64][]T,
+	loader ContractLoader[S],
```

Comment:
Is there actually any reason to have 2 separate functions? 
```
	loader ContractLoader[S],
	method ContractMethod[S, T],
```

I think it's okay now so we can get this shared functionality. But maybe in the next PR we condense this further so it's 1 function - and GetTxs wouldnt need to accept `loader`

Reply:
By separating the loader from the method you can reuse the same loader function for multiple methods on the same contract.